### DISCUSSION TITLE
Find closest point to pointer on a given path and draw a circle at the coordinates.

### DISCUSSION THREAD

**Discussion by jbwaring** [fabricjs/fabric.js#8441] [Q&A] [answered]
Dear FabricJS community, 

We are trying (on mouse:move) to have the pointer snap into the closest point of an SVG path. 
So far we are able to have the pointer snap into one of the four points of the bounding box of the closest object as shown here:
https://jsfiddle.net/b3qLxz07/46/ (example)

Is there a way to compute the coordinates of the closest point in the path with regards to another point? For the JSFiddle example, that would be a point on the triangle or any corners of the triangle but _not_ the bounding box. 

Thanks in advance for your suggestions, 

**Observed Behaviour**
https://user-images.githubusercontent.com/56231422/202032922-ea7dae47-6373-49c9-8448-fdc75767126b.mov

**Expected Behaviour**
We are able to find either a coordinate on the actual path which is closest to the pointer or the closest vertex.

Sincerely, 

JB

**Comment by ShaMan123** [comment, marked as answer]
@jbwaring your question isn't clear enough. Are you trying to snap the pointer to a triangle's vertexes? Are you trying to snap to a triangle's sides? Or are you trying to snap to any kind of path?

Answers respectively:
- ~~Use `triangle.getCoords()`~~
- compute triangle sides and compute distance of point from sides, then project to the closet side (you can look at [projectStroke](https://github.com/fabricjs/fabric.js/tree/master/src/util/misc/projectStroke) , [vectors](https://github.com/fabricjs/fabric.js/blob/master/src/util/misc/vectors.ts) for a direction)
- Use [path utils](https://github.com/fabricjs/fabric.js/blob/master/src/util/path.ts)
